### PR TITLE
Update nixUnstable + nixFlakes, remove nix1

### PR DESCRIPTION
**Pull request by edolstra**
###### Motivation for this change

Cleaned up the expression by removing `nix1`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
@GrahamcOfBorg eval

**Comment by grahamc** [comment]
It has a useful error now :)

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -75,6 +76,11 @@ common =
             patchelf --set-rpath $out/lib:${stdenv.cc.cc.lib}/lib $out/lib/libboost_thread.so.*
           ''}
         '' +
+        # Unpack the Rust crates.
+        ''
+          tar xvf ${crates} -C nix-rust/
```
Comment (cole-h) [inline pkgs/tools/package-management/nix/default.nix:81]: You probably already know this, but: since `crates` can be `null`, there should be a check here.

**Comment by cole-h** [comment]
Hope I'm not being too annoying: most recent eval failure is because you didn't also remove the `nix1` attribute from `top-level/all-packages.nix`. I'd link to the line, but GitHub doesn't load 25k lines; it's line 25446, or thereabouts.

**Comment by edolstra** [comment]
@cole-h Thanks, fixed!

**Comment by cole-h** [comment]
`nix run nixpkgs.nixpkgs-review -c nixpkgs-review pr 84214` results in the following:

```
[52 built (2 failed), 638 copied (3103.7 MiB), 473.9 MiB DL]
error: build of '/nix/store/cfzyv4f5hmyncazdv1mrl3grpr19i795-env.drv' failed
https://github.com/NixOS/nixpkgs/pull/84214
13 package marked as broken and skipped:
cadaver linuxPackages_hardkernel_4_14.mba6x_bl linuxPackages_hardkernel_4_14.nvidia_x11 linuxPackages_hardkernel_4_14.nvidiabl muslCross nix-delegate nix-deploy nix-exec nix-linter python37Packages.pythonnet python38Packages.libselinux python38Packages.nixpkgs-pytools python38Packages.pythonnet

2 package failed to build:
hydra-unstable python38Packages.nixpkgs

42 package built:
bundix cabal2nix cachix common-updater-scripts crystal2nix dep2nix disnix disnixos fusionInventory go2nix hydra-flakes hydra-migration lispPackages.quicklisp-to-nix lispPackages.quicklisp-to-nix-system-info nix nix-binary-cache nix-bundle nix-du nix-index nix-pin nix-plugins nix-prefetch nix-prefetch-bzr nix-prefetch-cvs nix-prefetch-docker nix-prefetch-git nix-prefetch-hg nix-prefetch-scripts nix-prefetch-svn nix-review nix-serve nix-update-source nixFlakes nixUnstable nixos-generators nixui pantheon.nixpkgs-github-update python37Packages.nixpkgs python37Packages.pythonix python38Packages.pythonix vgo2nix vulnix
```

Maybe a spurious failure on my system? Maybe safe to ignore? I dunno. But I'll leave this here in case it's useful.

<details><summary>hydra-unstable</summary>

```
builder for '/nix/store/sk2jwac7rk9hgcz9405amiqva5ihh73k-hydra-2020-03-24.drv' failed with exit code 2; last 10 log lines:
  hydra-eval-jobs.cc:158:38: error: base operand of '->' has non-pointer type 'nix::Attr'
    158 |                         auto v = (*a)->value->listElems()[n];
        |                                      ^~
  make[3]: *** [Makefile:440: hydra_eval_jobs-hydra-eval-jobs.o] Error 1
  make[3]: Leaving directory '/build/source/src/hydra-eval-jobs'
  make[2]: *** [Makefile:360: all-recursive] Error 1
  make[2]: Leaving directory '/build/source/src'
  make[1]: *** [Makefile:414: all-recursive] Error 1
  make[1]: Leaving directory '/build/source'
  make: *** [Makefile:344: all] Error 2
```
</details>

<details><summary>python38Packages.nixpkgs</summary>

```
builder for '/nix/store/avlyyyf7837na96321cd6v5wymnzir9f-python3.8-nixpkgs-0.2.4.drv' failed with exit code 1; last 10 log lines:
      import nix
  ModuleNotFoundError: No module named 'nix'


  ----------------------------------------------------------------------
  Ran 1 test in 0.000s

  FAILED (errors=1)
  Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
  error: Test failed: <unittest.runner.TextTestResult run=1 errors=1 failures=0>
```
</details>

**Comment by edolstra** [comment]
Looks like `hydra-unstable` should be updated (or removed). There are some fixes upstream.

Don't know what `python38Packages.nixpkgs` is.

**Pull request merged**